source/backend/dal/Repositories/ManagementActivityRepository.cs

```diff
@@ -0,0 +1,154 @@
+            {
+                predicate = predicate.And(x => EF.Functions.Like(x.ManagementFile.FileName, $"%{filter.FileNameOrNumberOrReference}%")
+                || EF.Functions.Like(x.ManagementFile.ManagementFileId.ToString(), $"%{filter.FileNameOrNumberOrReference}%")
+                || EF.Functions.Like(x.ManagementFile.LegacyFileNum, $"%{filter.FileNameOrNumberOrReference}%"));
+            }
+
+            if (!string.IsNullOrWhiteSpace(filter.ActivityTypeCode))
+            {
+                predicate = predicate.And(x => x.PropMgmtActivityTypeCode == filter.ActivityTypeCode);
+            }
+
+            if (!string.IsNullOrWhiteSpace(filter.ActivitySubTypeCode))
+            {
+                predicate = predicate.And(x => x.PropMgmtActivitySubtypeCode == filter.ActivitySubTypeCode);
+            }
+
+            if (!string.IsNullOrWhiteSpace(filter.ActivityStatusCode))
+            {
+                predicate = predicate.And(x => x.PropMgmtActivityStatusTypeCode == filter.ActivityStatusCode);
+            }
+
+            if (!string.IsNullOrWhiteSpace(filter.ProjectNameOrNumber))
+            {
+                predicate = predicate.And(x => EF.Functions.Like(x.ManagementFile.Project.Code, $"%{filter.ProjectNameOrNumber}%") || EF.Functions.Like(x.ManagementFile.Project.Description, $"%{filter.ProjectNameOrNumber}%"));
+            }
+
+            var query = Context.PimsPropertyActivities.AsNoTracking()
+                .Include(s => s.PropMgmtActivityStatusTypeCodeNavigation)
+                .Include(t => t.PropMgmtActivityTypeCodeNavigation)
+                .Include(st => st.PropMgmtActivitySubtypeCodeNavigation)
+                .Include(pp => pp.PimsPropPropActivities)
+                    .ThenInclude(p => p.Property)
+                        .ThenInclude(a => a.Address)
+                .Include(f => f.ManagementFile)
+                    .ThenInclude(pr => pr.PimsManagementFileProperties)
+                        .ThenInclude(p => p.Property)
+                            .ThenInclude(a => a.Address)
+
+                .Where(predicate);
+
+            if (filter.Sort?.Any() == true)
+            {
+                var field = filter.Sort.FirstOrDefault()?.Split(" ")?.FirstOrDefault();
+                var direction = filter.Sort.FirstOrDefault()?.Split(" ")?.LastOrDefault();
+
+                if (field == "ActivityStatus")
+                {
+                    query = direction == "asc" ? query.OrderBy(c => c.PropMgmtActivityStatusTypeCodeNavigation.Description) : query.OrderByDescending(c => c.PropMgmtActivityStatusTypeCodeNavigation.Description);
+                }
+                else if (field == "ActivityType")
+                {
+                    query = direction == "asc" ? query.OrderBy(c => c.PropMgmtActivityStatusTypeCodeNavigation.Description) : query.OrderByDescending(c => c.PropMgmtActivityStatusTypeCodeNavigation.Description);
```

Comment:
This looks like a copy/paste error. This should be the activity type not the status.